protocols/interfaces.go

```diff
@@ -9,24 +9,39 @@ import (
 	"github.com/statechannels/go-nitro/types"
 )
 
-// TransactionType is an enumeration of possible chain transactions
-type TransactionType string
-
-const (
-	DepositTransactionType     TransactionType = "Deposit"
-	WithdrawAllTransactionType TransactionType = "Withdraw"
-)
-
 var (
 	ErrNotApproved = errors.New("objective not approved")
 )
 
-// ChainTransaction is an object to be sent to a blockchain provider.
-type ChainTransaction struct {
-	// TODO support other transaction types (deposit, challenge, respond, conclude)
-	Type      TransactionType
-	ChannelId types.Destination
-	Deposit   types.Funds
+// ChainTransaction defines the interface that every transaction must implement
+type ChainTransaction interface {
+	ChannelId() types.Destination
+}
```

Comment:
I understand this is one of the ways we can build a sort of union type in Go. I don't have a concrete suggestion for something better right now... but I just want to flag the awkwardness of the fact that now `state.FixedPart` and `state.State` implement `protocols.ChainTransaction`.

Reply:
That's a good point. Down the line, the `ChainTransaction` interface will likely define more methods that are specific to chain transactions (as opposed to a generic `ChannelId()` methods).